src/apps/bridge/aanderaaController.cpp

```diff
@@ -240,11 +250,21 @@ static void runController(void *param) {
             // If not send NaNs for all the values.
             // TODO - verify that we can assume if one sampler is below the min then all of them are.
             if(curr->abs_speed_cm_s.getNumSamples() >= MIN_READINGS_FOR_AGGREGATION) {
-              agg.abs_speed_mean_cm_s = curr->abs_speed_cm_s.getMean(true);;
-              agg.abs_speed_std_cm_s = curr->abs_speed_cm_s.getStd(true);;
-              agg.direction_circ_mean_rad = curr->direction_rad.getCircularMean();;
-              agg.direction_circ_std_rad = curr->direction_rad.getCircularStd();
-              agg.temp_mean_deg_c = curr->temp_deg_c.getMean(true);;
+              if (curr->abs_speed_cm_s.getMean(true) > ABS_SPEED_SAMPLE_MIN && curr->abs_speed_cm_s.getMean(true) < ABS_SPEED_SAMPLE_MAX) {
+                agg.abs_speed_mean_cm_s = curr->abs_speed_cm_s.getMean(true);
+              }
```

Comment:
The functions like `getMean` do a potentially lengthy calculation. We don't want to call it 3 times, tripling the time complexity of this part of the code. Instead, call it once, saving the value in a variable, like the previous removed code would actually be fine. And then check that saved variable against the max and min, setting to NaN if outside.

Reply:
Good call! I will change it!